### DISCUSSION TITLE
When sync data with Flink, Doris uses an aggregation model, and the result of the sum field aggregation is incorrect.

### DISCUSSION THREAD

**Discussion by banyue0618** [apache/doris#30745] [A - General / Q&A] [answered]
过程是这样的：
mongodb数据增加一条，flink就会将数据同步至doris。
下面是sql语句（省略了source的定义sql）（这里为了测试，将数据直接打印，不存储至doris）。
custom_parse_json是我的udtf函数。

`select country,t1.`busivicode`, t1.cargname,
	case when t1.cargocatg is null then bas2.cargocatg else t1.cargocatg end as cargocatg , 
	case when t1.brandname is null then bas2.brandname else t1.brandname end as brandname, t1.predate
from (
	select country,province,city,channel,usercode,predate,`busivicode`,cargname,cargocatg,brandname from (
	select country, province, city, channel,usercode,predate,`busivicode`,`result` as queryresult from source_tpo_ogquery_log,unnest(`queryresults`) as t (`busivicode`,`result`) 
	WHERE queryresults IS NOT NULL and channel in ('10','20','30','40','50','60'))
	t0,LATERAL TABLE (custom_parse_json(`queryresult`,'gtsccode','cargname','cargocatg','brandname')) as t2 (gtsccode,cargname,cargocatg,brandname)) t1
left join (select p.portraitid,bas.cargname,bas.cargocatg,bas.brandname from source_dms_portraitid_dataid p 
			left join source_dms_cargo_baseinfo bas on p.cargosrcicode = bas.cargosrcicode) bas2 on (bas2.portraitid = t1.busivicode and bas2.cargname = t1.cargname);`

![image](https://github.com/apache/doris/assets/40993873/09b49924-0620-47fa-b46c-cca7859cb8df)

最后一条数据是mongodb新增的数据，flink将监听到的数据打印出来，只有一条。
整个过程，flink也只打印了一条数据。根据打印的数据条数这应该能说明实际上的数据只有一条【关于这点我并不能确定】。
于是有了下面的问题
doris数据入库之后的sum字段数值应该是1，但是结果却是5。说明在doris里新增了五条一样的数据！！！.
关于环境：我的flink都是单机模式，doris的be也是单节点。
这是我的doris建表语句。
![image](https://github.com/apache/doris/assets/40993873/24edfe5d-aa52-4536-a3e7-65c28e1bbec2)

无法理解这个现象！请问可以帮助我吗。

**Comment by zhbdesign** [comment]
你先用明细模型表，看到底进去的是几条

**Reply by banyue0618** [reply]
确实得用明细模型，我才发现业务中，一条查询的日志记录入库之后，还会更新四次（这个业务逻辑我是万万没想到）！！！，导致flink监听到五条（但是纳闷为啥这里只打印了一条记录，这是bug吗？）。最终导致sum的字段数量不对。
我如果还是想用聚合模型的话，那就必须mongodb-cdc配置只监听insert事件，可是我今天找了一个中午我都没找到相关的配置，貌似是不支持的。
![image](https://github.com/apache/doris/assets/40993873/72a77d8c-f1f3-40c4-b6c9-b09649940e50)

可是，以后数据量大了（达到亿级再往上），明细模型肯定会有性能上的问题。这可如何是好。

**Reply by zhbdesign** [reply]
可以物化视图加速，你可以只监听mongo，然后打印，看看insert，update是几条，你这sql都做了处理了，我自己用的mysqlcdc，对于update，mysql出来的op类型是d和i两条记录

**Reply by banyue0618** [reply]
你好，听你的意思是，使用cdc同步的时候可以拿到数据的操作类型，可是我再mysql-cdc中找了很久都没有找到相关配置，因为如果能配置过滤掉更新及删除动作，最终只同步insert的数据那最好不过了。
采用物化视图优化：使用物化视图，可是我count列需要进行去重，按照官网介绍，得使用函数bitmap_union(to_bitmap(列名))，但是我这个列是varchar类型，那物化视图也使用不了，看起来去重列只能是int类型。

**Reply by zhbdesign** [reply, marked as answer]
operation     STRING METADATA FROM 'row_kind' VIRTUAL,这是mysqlcdc可以获取的元数据，https://ververica.github.io/flink-cdc-connectors/master/content/connectors/mysql-cdc%28ZH%29.html

**Reply by banyue0618** [reply]
原来是这样的，一直以为是在配置列里面，感谢感谢。
